lib/openactive/base_model.rb

```diff
@@ -0,0 +1,164 @@
+module OpenActive
+  class BaseModel
+    include OpenActive::Concerns::JsonLdSerializable
+    include OpenActive::Concerns::TypeChecker
+    # class << self
+    #   extend OpenActive::Concerns::TypeChecker::PrependedClassMethods
+    # end
+
+    def initialize(attributes = {})
+      assign_attributes(attributes) if attributes
+
+      super()
+    end
+
+    def assign_attributes(attributes)
+      # attributes.each do |k, v|
+      #   setter = "#{k}="
+      #   public_send(setter, v) if respond_to?(setter)
+      # end
+
+      attributes.each do |key, value|
+        set_property(key, value)
+      end
+    end
+
+    def self.define_property(property, types: nil, as: nil)
+      attr_accessor property
+      property property, as: as if as
+      validate_property property, types: types if types
+    end
+
+    # Gets or sets the identifier used to uniquely identify things that are being described in the document with
+    # IRIs or blank node identifiers.
+    # To be able to externally reference nodes in a graph, it is important that nodes have an identifier. IRIs
+    # are a fundamental concept of Linked Data, for nodes to be truly linked, dereferencing the identifier should
+    # result in a representation of that node.This may allow an application to retrieve further information about
+    # a node. In JSON-LD, a node is identified using the @id keyword:
+    #
+    # @var string
+    attr_accessor :id
+    property :id, as: "id"
+
+    # Gets the context used to define the short-hand names that are used throughout a JSON-LD document.
+    # These short-hand names are called terms and help developers to express specific identifiers in a compact
+    # manner.
+    # When two people communicate with one another, the conversation takes place in a shared environment,
+    # typically called "the context of the conversation". This shared context allows the individuals to use
+    # shortcut terms, like the first name of a mutual friend, to communicate more quickly but without losing
+    # accuracy. A context in JSON-LD works in the same way. It allows two applications to use shortcut terms to
+    # communicate with one another more efficiently, but without losing accuracy.
+    # Simply speaking, a context is used to map terms to IRIs. Terms are case sensitive and any valid string that
+    # is not a reserved JSON-LD keyword can be used as a term.
+    #
+    # @var string[]
+    attr_accessor :context
+
+    def context
+      @context ||= [
+        "https://openactive.io/",
+        "https://openactive.io/ns-beta"
+      ]
+    end
+
+    def set_property(key, value)
+      attr_name = key.to_s.underscore
+
+      attr_name = attr_name[1..] if attr_name.start_with?('@')
+
+      val = value
+
+      if value.is_a?(Array) || value.is_a?(Hash)
+        val = deserialize_value(value)
+        # elsif value.is_a?(BaseModel)
+        #   val = value.class.deserialize(value)
```

Comment:
Remove commented out code

Reply:
Done.

Reply:
I think you may not have pushed this removal ;-)